lib/pure/httpclient.nim

```diff
@@ -387,13 +395,30 @@ proc getDefaultSSL(): SslContext =
       result = defaultSslContext
       doAssert result != nil, "failure to initialize the SSL context"
 
-proc newProxy*(url: string; auth = ""): Proxy =
+proc newProxy*(url: Uri): Proxy =
   ## Constructs a new `TProxy` object.
-  result = Proxy(url: parseUri(url), auth: auth)
+  result = Proxy(url: url)
 
-proc newProxy*(url: Uri; auth = ""): Proxy =
+proc newProxy*(url: string): Proxy =
   ## Constructs a new `TProxy` object.
-  result = Proxy(url: url, auth: auth)
```

Comment:
Likewise for all these constructors.

Reply:
The code is not broken, I've introduced backward compatible (but deprecated) constructors and auth getter. I've removed the field, because of ambiguity with url, which IMO is a well-established format for providing/storing username/password.